test/src/service/v1/notifications/notifications_v1_service_test.dart

```diff
@@ -49,70 +50,218 @@ void main() {
         ),
       );
 
-      final response = await notificationsService.clearAllNotifications();
+      final response = await notificationsService.destroyAllNotifications();
 
       expect(response, isA<MastodonResponse>());
       expect(response.rateLimit, isA<RateLimit>());
       expect(response.data, isTrue);
     });
+
+    test('when unauthorized', () async {
+      final notificationsService = NotificationsV1Service(
+        instance: 'test',
+        context: context.buildPostStub(
+          'test',
+          UserContext.oauth2Only,
+          '/api/v1/notifications/clear',
+          'test/src/service/v1/notifications/data/destroy_all_notifications.json',
+          statusCode: 401,
+        ),
+      );
+
+      expectUnauthorizedException(
+        () async => await notificationsService.destroyAllNotifications(),
+      );
+    });
+
+    test('when rate limit exceeded', () async {
+      final notificationsService = NotificationsV1Service(
+        instance: 'test',
+        context: context.buildPostStub(
+          'test',
+          UserContext.oauth2Only,
+          '/api/v1/notifications/clear',
+          'test/src/service/v1/notifications/data/destroy_all_notifications.json',
+          statusCode: 429,
+        ),
+      );
+
+      expectRateLimitExceededException(
+        () async => await notificationsService.destroyAllNotifications(),
+      );
+    });
   });
 
-  group('.clearNotificationById', () {
+  group('.destroyNotification', () {
     test('normal case', () async {
       final notificationsService = NotificationsV1Service(
         instance: 'test',
         context: context.buildPostStub(
           'test',
           UserContext.oauth2Only,
           '/api/v1/notifications/12345/dismiss',
-          'test/src/service/v1/notifications/data/clear_notification_by_id.json',
+          'test/src/service/v1/notifications/data/destroy_notification.json',
         ),
       );
 
-      final response = await notificationsService.clearNotificationById(
+      final response = await notificationsService.destroyNotification(
         notificationId: '12345',
       );
 
       expect(response, isA<MastodonResponse>());
       expect(response.rateLimit, isA<RateLimit>());
       expect(response.data, isTrue);
     });
+
+    test('when unauthorized', () async {
+      final notificationsService = NotificationsV1Service(
+        instance: 'test',
+        context: context.buildPostStub(
+          'test',
+          UserContext.oauth2Only,
+          '/api/v1/notifications/12345/dismiss',
+          'test/src/service/v1/notifications/data/destroy_notification.json',
+          statusCode: 401,
+        ),
+      );
+
+      expectUnauthorizedException(
+        () async => await notificationsService.destroyNotification(
+          notificationId: '12345',
+        ),
+      );
+    });
+
+    test('when rate limit exceeded', () async {
```

Comment:
Hey, @myConsciousness why do you need every time test rate limit and authorization response for each request?
I see that is stubbed data which comes from BaseService and all method use that solution.
An endpoint methods should test for available parameters and expected response object.
Did I miss something? What cases can such a case cover?

Reply:
Hi @Kraigo ,

These are tests to ensure that each endpoint can reliably handle exception cases :) An exception case that endpoint A can handle perfectly and endpoint B returns a different exception or error would create chaos.

The processing in this package is commonized like a framework, but it’s up to the implementer to be able to use and implement it correctly. So, even if the code is redundant, these common specifications must be tested to ensure their correctness :)